pallets/currencies/src/lib.rs

```diff
@@ -960,6 +960,12 @@ where
 	}
 }
 
+impl<T: Config> GetByKey<CurrencyIdOf<T>, BalanceOf<T>> for Pallet<T> {
```

Comment:
Create a struct AssetTotalIssuance and implement Get by key for that structure, instead of implementing it for Pallet.